web-app/js/portal/search/FreeTextSearchPanel.js

```diff
@@ -23,14 +23,29 @@ Portal.search.FreeTextSearchPanel = Ext.extend(Ext.Panel, {
                     items: [
                         {
                             html: '<span class=\"fa fa-search \"></span>',
-                            cls: 'fa fa-2x'
+                            cls: 'fa fa-2x',
+                            scope: this,
+                            listeners: {
+                                render: function(c) {
+                                    c.body.on('click', function() {
+                                        trackFacetUsage(this.facetName, OpenLayers.i18n('facetTrackingFreeTextMagnifierLabel'));
+                                    });
+                                }
+                            }
                         },
                         { xtype: 'spacer', width: 10 },
                         this.searchField = new Ext.form.TextField({
                             width: 250,
                             focusClass: "focused",
                             emptyText: OpenLayers.i18n('freeTextFilterSearchHelperMsg'),
                             fieldLabel: 'Name',
+                            listeners: {
+                                specialkey: function(field, e) {
+                                    if (e.getKey() == e.ENTER) {
+                                        trackFacetUsage(this.facetName, OpenLayers.i18n('facetTrackingFreeTextEnterLabel'));
```

Comment:
Is `facetName` accessible from here? When I replace `trackFacetUsage` with `console.log`, I get "undefined"